ext/har/types.go

```diff
@@ -0,0 +1,414 @@
+    
+    contentType := req.Header.Get("Content-Type")
+    if contentType == "" {
+        return nil
+    }
+    
+    mediaType, _, err := mime.ParseMediaType(contentType)
+    if err != nil {
+        log.Printf("Error parsing media type: %v", err)
+        return nil
+    }
+    
+    harPostData.MimeType = mediaType
+    
+    if err := req.ParseForm(); err != nil {
+        log.Printf("Error parsing form: %v", err)
+        return nil
+    }
+    
+    if len(req.PostForm) > 0 {
+        for k, vals := range req.PostForm {
+            for _, v := range vals {
+                param := PostDataParam{
+                    Name:  k,
+                    Value: v,
+                }
+                harPostData.Params = append(harPostData.Params, param)
+            }
+        }
+    } else {
+        str, err := io.ReadAll(req.Body)
+        if err != nil {
+            log.Printf("Error reading request body: %v", err)
+            return nil
+        }
+        harPostData.Text = string(str)
+    }
+    
+    return harPostData
+}
+
+func parseStringArrMap(stringArrMap map[string][]string) []NameValuePair {
+	harQueryString := make([]NameValuePair, 0, len(stringArrMap))
+	
+	for k, v := range stringArrMap {
+		escapedKey, err := url.QueryUnescape(k)
+		if err != nil {
+			// Use original key if unescaping fails
+			escapedKey = k
+		}
+
+		escapedValues, err := url.QueryUnescape(strings.Join(v, ","))
+		if err != nil {
+			// Use original joined values if unescaping fails
+			escapedValues = strings.Join(v, ",")
+		}
+
+		harNameValuePair := NameValuePair{
+			Name:  escapedKey,
+			Value: escapedValues,
+		}
+		
+		harQueryString = append(harQueryString, harNameValuePair)
+	}
+	
+	return harQueryString
+}
+
+func parseCookies(cookies []*http.Cookie) []Cookie {
+	harCookies := make([]Cookie, len(cookies))
+	for i, cookie := range cookies {
+		harCookie := Cookie{
+			Name:     cookie.Name,
+			Domain:   cookie.Domain,
+			HttpOnly: cookie.HttpOnly,
+			Path:     cookie.Path,
+			Secure:   cookie.Secure,
+			Value:    cookie.Value,
+		}
+		if !cookie.Expires.IsZero() {
+			harCookie.Expires = &cookie.Expires
+		}
+		harCookies[i] = harCookie
+	}
+	return harCookies
+}
+
+type Response struct {
+	Status      int             `json:"status"`
+	StatusText  string          `json:"statusText"`
+	HttpVersion string          `json:"httpVersion"`
+	Cookies     []Cookie        `json:"cookies"`
+	Headers     []NameValuePair `json:"headers"`
+	Content     Content         `json:"content"`
+	RedirectUrl string          `json:"redirectURL"`
+	BodySize    int64           `json:"bodySize"`
+	HeadersSize int64           `json:"headersSize"`
+	Comment     string          `json:"comment,omitempty"`
+}
+
+func ParseResponse(resp *http.Response, captureContent bool) *Response {
+    if resp == nil {
+        return nil
+    }
+
+    statusText := resp.Status
+    if len(resp.Status) > 4 {
+        statusText = resp.Status[4:]
+    }
```

Comment:
Instead of this manual stuff, use http.StatusText(resp.StatusCode), directly inside the Response{} creation in the next lines